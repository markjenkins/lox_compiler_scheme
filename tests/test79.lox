// adapted from craftinginterpreters/test/assignment/syntax.lox

// Assignment on RHS of variable.
{
    var a = "before";
    var c = a = "var";
    print a; // expect: var
    print c; // expect: var
}
