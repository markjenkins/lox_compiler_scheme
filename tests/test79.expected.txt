OP_CONSTANT "before"
OP_CONSTANT "var"
OP_SET_LOCAL 0
OP_GET_LOCAL 0
OP_PRINT
OP_GET_LOCAL 1
OP_PRINT
OP_POP
OP_POP
OP_RETURN
